src/model.rs

```diff
@@ -1784,6 +1787,13 @@ mod tests {
             );
             collections.vehicle_journeys =
                 build_vehicle_journeys(stop_config, next_vj_config_config);
+            let mut dates = std::collections::BTreeSet::new();
+            dates.insert(Date::from_ymd(2020, 1, 1));
```

Comment:
Don't know if it's better, but for oneliner const:
```suggestion
            let dates: std::collections::BTreeSet::new() = [Date::from_ymd(2020, 1, 1)].iter().cloned().collect();
```

Reply:
Nice one, but I'm not sure how to avoid additional allocation. Because with your solution, you have `.cloned()` which means the `Date` will exist twice (once on the stack, and once on the heap). And an alternative solution I had will allocate a `Vec`.

```rust
let dates: std::collections::BTreeSet<_> = vec![Date::from_ymd(2020, 1, 1)].into_iter().collect();
```
In the end, the solution from @antoine-de is probably the most efficient one. In one of my side project, I also did the following (but that need an additional macro).

```rust
macro_rules! btree_set {
    ($($e:expr),*) => {{
        let mut btree_set = std::collections::BTreeSet::new();
        $(btree_set.insert($e);)*
        btree_set
    }}
}
// ...
let dates = btree_set![Date::from_ymd(2020, 1, 1)];
```

But adding a new macro if it's used only once...

Reply:
a date is basicaly an `i32`, copying is very cheap (and we don't care much about perf here I think)
I think we should be driven by expressiveness here, I'd vote for the moment the one liner as think it's quite readable, and if we have to do it too many time, we add the macro, is that ok for you?

Reply:
It's ok for me, you can replace the `.cloned()` with a `.copied()` to better express it :wink: